packages/react-tooltip/src/components/Tooltip/useTooltip.tsx

```diff
@@ -30,53 +38,214 @@ export const useTooltip = (
+      const delay = context.hideVisibleTooltip ? 0 : state.showDelay;
+
+      setDelayTimeout(() => {
+        if (hoveredOrFocused.current) {
+          setVisible(true);
+        }
+      }, delay);
+    },
+    [context, popper.targetRef, setDelayTimeout, state.onlyIfTruncated, state.showDelay, state.targetRef],
+  );
+
+  // Listener for onPointerLeave and onBlur on the trigger element
+  const onLeaveTrigger = React.useCallback(() => {
+    hoveredOrFocused.current = false;
+
+    setDelayTimeout(() => {
+      if (!hoveredOrFocused.current) {
+        setVisible(false);
+      }
+    }, state.hideDelay);
+  }, [setDelayTimeout, state.hideDelay]);
+
+  // Listen for the mouse entering/leaving the tooltip, and treat it as if hovered over the trigger.
+  // This keeps the tooltip visible when the pointer is moved over it.
+  const { onPointerEnter, onPointerLeave } = state;
+  state.onPointerEnter = ev => {
+    hoveredOrFocused.current = true;
+    onPointerEnter?.(ev);
+  };
+  state.onPointerLeave = ev => {
+    onLeaveTrigger();
+    onPointerLeave?.(ev);
+  };
+
+  const childProps = React.isValidElement(state.children) ? state.children.props : undefined;
+
+  // The props to add to the trigger element (child)
+  const triggerProps: TooltipTriggerProps = {
+    onPointerEnter: useMergedCallbacks(childProps?.onPointerEnter, onEnterTrigger),
+    onPointerLeave: useMergedCallbacks(childProps?.onPointerLeave, onLeaveTrigger),
+    onFocus: useMergedCallbacks(childProps?.onFocus, onEnterTrigger),
+    onBlur: useMergedCallbacks(childProps?.onBlur, onLeaveTrigger),
+  };
 
-  const mergedArrowRef = useMergedRefs(state.arrow?.ref, arrowRef);
-  if (state.arrow) {
-    state.arrow.ref = mergedArrowRef;
+  if (state.type === 'description') {
+    if (state.shouldRenderTooltip) {
+      triggerProps['aria-describedby'] = state.id;
+    }
+  } else if (typeof state.content.children !== 'string') {
+    if (state.shouldRenderTooltip) {
+      triggerProps['aria-labelledby'] = state.id;
+    }
+  } else {
+    // If the trigger's label is a simple string, then we can use the aria-label prop, and
+    // we don't need to render the tooltip content when it isn't visible
+    triggerProps['aria-label'] = state.content.children as string;
+    if (!visible) {
+      state.shouldRenderTooltip = false;
+    }
+  }
+
+  // Apply the trigger props to the child, either by calling the render function, or cloning with the new props
+  if (typeof state.children === 'function') {
+    state.children = state.children(triggerProps) as TooltipState['children'];
+  } else {
+    state.children = React.cloneElement(onlyChild(state.children), triggerProps);
   }
 
   return state;
 };
+
+/**
+ * Helper to manage a browser timeout.
+ * Ensures that the timeout isn't set multiple times at once,
+ * and is cleaned up when the component is unloaded.
+ *
+ * @returns A pair of [setTimeout, clearTimeout] that are stable between renders.
+ */
+const useTimeout = () => {
```

Comment:
Might be worth moving to `@fluentui/react-hooks` but can be done in a separate PR.

Reply:
Yeah it could be useful as a shared utility. I'll do that in a separate PR.